src/lib/free-queue/test/free-queue.test.js

```diff
@@ -0,0 +1,162 @@
+      const testData = [new Float32Array(bufferLength * 2).fill(1), new Float32Array(bufferLength * 2).fill(2)];
+      freeQueue.push(testData);
+
+      expect(freeQueue.framesAvailable).to.equal(bufferLength);
+    });
+
+    it('should not pull data when buffer is empty', () => {
+      const outputData = [new Float32Array(bufferLength), new Float32Array(bufferLength)];
+      freeQueue.pull(outputData);
+
+      expect(outputData[0]).to.deep.equal(new Float32Array(bufferLength));
+      expect(outputData[1]).to.deep.equal(new Float32Array(bufferLength));
+    });
+
+    it('should manage partial data pulls', () => {
+      const testData = [new Float32Array(bufferLength).fill(1), new Float32Array(bufferLength).fill(2)];
+      freeQueue.push(testData);
+
+      const partialOutput = [new Float32Array(bufferLength / 2), new Float32Array(bufferLength / 2)];
+      freeQueue.pull(partialOutput);
+
+      expect(partialOutput[0]).to.deep.equal(new Float32Array(bufferLength / 2).fill(1));
+      expect(partialOutput[1]).to.deep.equal(new Float32Array(bufferLength / 2).fill(2));
+      expect(freeQueue.framesAvailable).to.equal(bufferLength / 2);
+    });
+
+    it('should handle multiple push and pull cycles', () => {
+      const testData = [new Float32Array(bufferLength).fill(1), new Float32Array(bufferLength).fill(2)];
+      
+      for (let i = 0; i < 5; i++) {
+        freeQueue.push(testData);
+
+        const outputData = [new Float32Array(bufferLength), new Float32Array(bufferLength)];
+        freeQueue.pull(outputData);
+
+        expect(outputData[0]).to.deep.equal(testData[0]);
+        expect(outputData[1]).to.deep.equal(testData[1]);
+        expect(freeQueue.framesAvailable).to.equal(0);
+      }
+    });
+  });
+
+  describe('Error Handling', () => {
+    it('should return null for invalid channel index in getChannelData', () => {
+      const invalidIndex = channelCount + 1;
+      expect(freeQueue.getChannelData(invalidIndex)).to.be.null;
+    });
+
+    it('should throw an error if pushing with mismatched channel count', () => {
+      const invalidTestData = [new Float32Array(bufferLength).fill(1)];
+      
+      const expectedChannelCount = freeQueue._channelCount; 
+      const actualChannelCount = invalidTestData.length;
+    
+      expect(() => freeQueue.push(invalidTestData))
+        .to.throw(Error, `Channel count mismatch: expected ${expectedChannelCount}, but got ${actualChannelCount}.`);
+    });
+
+    it('should throw an error if pulling with mismatched channel count', () => {
+      const invalidOutputData = [new Float32Array(bufferLength)]; 
+      
+      const expectedChannelCount = freeQueue._channelCount; 
+      const actualChannelCount = invalidOutputData.length;
+    
+      expect(() => freeQueue.pull(invalidOutputData))
+        .to.throw(Error, `Channel count mismatch: expected ${expectedChannelCount}, but got ${actualChannelCount}.`);
+    });
+  });
+
+  describe('Performance', () => {
```

Comment:
Let's remove these and port the large and small data tests from FreeQueueSAB tests here instead.